qt/widgets/common/src/SequentialFitDialog.cpp

```diff
@@ -373,7 +372,64 @@ void SequentialFitDialog::functionChanged() { populateParameters(); }
 
 void SequentialFitDialog::finishHandle(
     const Mantid::API::IAlgorithm * /*alg*/) {
+  getFitResults();
   emit needShowPlot(&ui, m_fitBrowser);
+  m_fitBrowser->sequentialFitFinished();
+}
+
+/// Set the parameters to the fit outcome.
+/// if more than one entry in the gui the parameters used are first entry with
+/// same name as fit property browser if one entry will try and use row that
+/// corresponds to ws index in fit browser, if not found defaults to first row.
+void SequentialFitDialog::getFitResults() {
+  auto wsName = m_outputName;
+  if (!Mantid::API::AnalysisDataService::Instance().doesExist(wsName)) {
+    return;
+  }
+  Mantid::API::ITableWorkspace_sptr ws =
+      boost::dynamic_pointer_cast<Mantid::API::ITableWorkspace>(
+          Mantid::API::AnalysisDataService::Instance().retrieve(wsName));
+  if (!ws) {
+    return;
+  }
+  auto columnNames = ws->getColumnNames();
+
+  size_t rowNo = 0;
+  if (rowCount() > 1) {
+    // first column contains ws names
+    auto firstColumn = ws->getColumn(0);
+    for (size_t i = 0; i < ws->rowCount(); ++i) {
+      if (firstColumn->cell<std::string>(i) == m_fitBrowser->workspaceName())
```

Comment:
I think there is a bug here, as the if does not have braces
Currently you break on the first iteration of the for loop

Reply:
Ah good catch - I've added the brackets in so it should be correct now